lazyflow/utility/io_util/tiledVolume.py

```diff
@@ -325,42 +317,30 @@ def _retrieve_remote_tile(self, rest_args, tile_relative_intersection, data_out)
-                self._session = self._create_session()
-
-                # Provide authentication if we have the details.
-                if self.description.username and self.description.password:
-                    self._session.auth = (self.description.username, self.description.password)
-
-            success = False
-            tries = 0
-            while not success:
-                try:
-                    # Note: We give timeout as a tuple, which requires a recent version of requests.
-                    #       If you get an exception about that, upgrade your requests module.
-                    r = self._session.get(tile_url, timeout=(3.0, 20.0))
-                    success = True
-                except requests.ConnectionError:
-                    # This special 'pass' is here because we keep running into exceptions like this:
-                    #   ConnectionError: HTTPConnectionPool(host='neurocean.int.janelia.org', port=6081):
-                    #   Max retries exceeded with url: /ssd-3-tiles/abd1.5/43/24_25_0.jpg
-                    #   (Caused by <class 'httplib.BadStatusLine'>: '')
-                    # So now we loop a few times and only give up if something is really wrong.
-                    if tries == 5:
-                        raise  # give up
-                    tries += 1
-        except:
-            # During testing, the server we're pulling from might be in our own process.
-            # Apparently that means that it is not very responsive, leading to exceptions.
-            # As a cheap workaround, just try one more time.
-            if self.TEST_MODE:
-                import time
-
-                time.sleep(0.01)
+        if self._session is None:
+            self._session = self._create_session()
+
+            # Provide authentication if we have the details.
+            if self.description.username and self.description.password:
+                self._session.auth = (self.description.username, self.description.password)
+
+        success = False
+        tries = 0
+        while not success:
+            try:
+                # Note: We give timeout as a tuple, which requires a recent version of requests.
+                #       If you get an exception about that, upgrade your requests module.
                 r = self._session.get(tile_url, timeout=(3.0, 20.0))
-            else:
-                raise
+                success = True
+            except requests.ConnectionError:
+                # This special 'pass' is here because we keep running into exceptions like this:
+                #   ConnectionError: HTTPConnectionPool(host='neurocean.int.janelia.org', port=6081):
+                #   Max retries exceeded with url: /ssd-3-tiles/abd1.5/43/24_25_0.jpg
+                #   (Caused by <class 'httplib.BadStatusLine'>: '')
+                # So now we loop a few times and only give up if something is really wrong.
+                if tries == 5:
```

Comment:
I think it would be useful to make this configurable (how many retries) more prominently.

Reply:
Removed custom retries, just using normal requests ones